nucypher/cli/deploy.py

```diff
@@ -60,96 +67,243 @@ def deploy(click_config,
+    ETH_NODE = None
+
+    #
+    # Validate
+    #
+
+    # Ensure config root exists, because we need a default place to put output files.
     config_root = config_root or DEFAULT_CONFIG_ROOT
     if not os.path.exists(config_root):
         os.makedirs(config_root)
 
-    # Establish a contract Registry
+    #
+    # Connect to Blockchain
+    #
+
+    # Establish a contract registry from disk if specified
     registry, registry_filepath = None, (registry_outfile or registry_infile)
     if registry_filepath is not None:
         registry = EthereumContractRegistry(registry_filepath=registry_filepath)
 
-    # Connect to Blockchain
+    if geth:
+        # Spawn geth child process
+        ETH_NODE = NuCypherGethDevnetProcess(config_root=config_root)
+        ETH_NODE.ensure_account_exists(password=click_config.get_password(confirm=True))
+        if not ETH_NODE.initialized:
+            ETH_NODE.initialize_blockchain()
+        ETH_NODE.start()  # TODO: Graceful shutdown
+        provider_uri = ETH_NODE.provider_uri
+
+    # Deployment-tuned blockchain connection
     blockchain = Blockchain.connect(provider_uri=provider_uri,
+                                    poa=poa,
                                     registry=registry,
-                                    deployer=True,
                                     compile=not no_compile,
-                                    poa=poa)
+                                    deployer=True,
+                                    fetch_registry=False,
+                                    sync=sync)
+
+
+    #
+    # Deployment Actor
+    #
 
-    # OK - Let's init a Deployment actor
     if not deployer_address:
-        etherbase = blockchain.interface.w3.eth.accounts[0]
-        deployer_address = etherbase  # TODO: Make this required instead, perhaps interactive
+        for index, address in enumerate(blockchain.interface.w3.eth.accounts):
+            click.secho(f"{index} --- {address}")
+        choices = click.IntRange(0, len(blockchain.interface.w3.eth.accounts))
+        deployer_address_index = click.prompt("Select deployer address", default=0, type=choices)
+        deployer_address = blockchain.interface.w3.eth.accounts[deployer_address_index]
 
-    click.confirm("Deployer Address is {} - Continue?".format(deployer_address), abort=True)
+    # Verify Address
+    if not force:
+        click.confirm("Selected {} - Continue?".format(deployer_address), abort=True)
     deployer = Deployer(blockchain=blockchain, deployer_address=deployer_address)
 
-    # The Big Three
-    if action == "contracts":
-        secrets = click_config.collect_deployment_secrets()
-
-        # Track tx hashes, and new agents
-        __deployment_transactions = dict()
-        __deployment_agents = dict()
+    # Verify ETH Balance
+    click.secho(f"\n\nDeployer ETH balance: {deployer.eth_balance}")
+    if deployer.eth_balance is 0:
```

Comment:
I think it's very risky to test equality of numerical quantities with `is`. This has been done several times (see #975, e.g.)

Reply:
Yes, agreed.  I think this is a refactoring artifact.

Reply:
:heavy_check_mark: